### PR TITLE
test(e2e): patch installplans w/ server-side apply

### PR DESCRIPTION
**Pull request by njhale**
Use server-side apply to patch InstallPlans under test in order to prevent
update conflict flakes.

### CODE REVIEW

**Comment by njhale** [comment]
/hold

Looks like the patch is still running into issues with `resourceVersion`.

File: test/e2e/subscription_e2e_test.go [hunk 1]
```diff
@@ -948,106 +944,106 @@ var _ = Describe("Subscription", func() {
...
-		require.NoError(GinkgoT(), err)
+		plan, err := crc.OperatorsV1alpha1().InstallPlans(ref.Namespace).Get(context.Background(), ref.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
 		// Set the InstallPlan's approval mode to Manual
 		plan.Spec.Approval = v1alpha1.ApprovalManual
 		plan.Spec.Approved = false
-		plan, err = crc.OperatorsV1alpha1().InstallPlans(ref.Namespace).Update(context.TODO(), plan, metav1.UpdateOptions{})
-		require.NoError(GinkgoT(), err)
+		setTypeMeta := func(p *v1alpha1.InstallPlan) *v1alpha1.InstallPlan {
```
Comment (ecordell) [inline test/e2e/subscription_e2e_test.go:962]: if this is in an eventually, it should work?

**Comment by exdx** [comment]
This is really interesting. Does it make any sense to consider turning this logic into a library such that more of the e2e tests can start to use server-side apply?

**Comment by njhale** [comment]
> This is really interesting. Does it make any sense to consider turning this logic into a library such that more of the e2e tests can start to use server-side apply?

@exdx The function is included with our e2e tests here, so it's already available to use in any of them.

File: test/e2e/util_test.go [hunk 2]
```diff
@@ -882,3 +884,104 @@ func toggleFeatureGates(deployment *appsv1.Deployment, toToggle ...featuregate.F
...
+		panic(fmt.Sprintf("unable to extract key from resource: %s", err))
+	}
+
+	return func() error {
+		cp := obj.DeepCopyObject().(Object)
+		if err := client.Get(bg, key, cp); err != nil {
+			return err
+		}
+		cp.GetObjectKind().SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		cp.SetManagedFields(nil)
```
Comment (ecordell) [inline test/e2e/util_test.go:946]: is this needed? shouldn't the `ForceOwnership` make `test` become the owner of any changed fields?

Comment (njhale) [reply test/e2e/util_test.go:952]: this came out of some red/green/refactor-ing -- an error is returned without it.

File: test/e2e/util_test.go [hunk 3]
```diff
@@ -882,3 +884,110 @@ func toggleFeatureGates(deployment *appsv1.Deployment, toToggle ...featuregate.F
...
+		out = change.Call([]reflect.Value{reflect.ValueOf(cp)})
+		if err := out[0].Interface(); err != nil {
+			return err.(error)
+		}
+
+		if err := client.Status().Patch(bg, cp, controllerclient.Apply, controllerclient.ForceOwnership, controllerclient.FieldOwner("test")); err != nil {
+			return err
+		}
+
+		reflect.ValueOf(obj).Elem().Set(cpv.Elem())
```
Comment (benluddy) [inline test/e2e/util_test.go:989]: Does the doc comment make it clear that the value addressed by`obj` will be updated?

**Pull request merged**